lib/compress.js

```diff
@@ -601,12 +601,18 @@ merge(Compressor.prototype, {
                             ret = funs.concat([ stat.transform(compressor) ]);
                             continue loop;
                         }
+
                         //---
-                        // XXX: what was the intention of this case?
+                        // function f(e){if (a(e)) return foo();if (b(e)) return bar();if (c(e)) return baz();fail(e)};
+                        // ==> function f(e){return a(e)?foo():b(e)?bar():c(e)?baz():void fail(e)}
+                        //
                         // if sequences is not enabled, this can lead to an endless loop (issue #866).
                         // however, with sequences on this helps producing slightly better output for
                         // the example code.
+                        //
+                        // Also require to at least have 2 return statements to counter the cost of the added `void `
                         if (compressor.option("sequences")
+                            && i > 0 && statements[i - 1] instanceof AST_If && statements[i - 1].body instanceof AST_Return
```

Comment:
In case that there is no if statement above, there is only 1 return statement. This case will add a `void` in front of the tailing last statement. This cost must be compensated.

There might be a case that there is a `return` statement above, but this is dead code handling.